libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -109,6 +109,10 @@ func defaultConfig(settings report.Settings) config {
 		c.Format = settings.Format
 	}
 
+	if settings.OverrideClusterUUID != "" {
+		c.OverrideClusterUUID = settings.OverrideClusterUUID
+	}
```

Comment:
there is no default uuid set in c. The check and assignment is superfluous. Better set this in the initializer directly then.

Reply:
Sorry if I'm missing something obvious here but I didn't quite follow your suggestion.

> there is no default uuid set in c. The check and assignment is superfluous. 

The reason I ended up with this check and assignment here is:

1. Ultimately I need to use `clusterUUID` here: https://github.com/ycombinator/beats/blob/1df1cf41d1de7fde07472e602888d26199221d0d/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L312

2. ... which is passed into the function here: https://github.com/ycombinator/beats/blob/1df1cf41d1de7fde07472e602888d26199221d0d/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L273

3. ... which is called here: https://github.com/ycombinator/beats/blob/1df1cf41d1de7fde07472e602888d26199221d0d/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L268

   and here: https://github.com/ycombinator/beats/blob/1df1cf41d1de7fde07472e602888d26199221d0d/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L270

4. Both the above calls are made from the `(r *reporter) initLoop(c config)` function, which is called with an `elasticsearch.config` object here: https://github.com/ycombinator/beats/blob/1df1cf41d1de7fde07472e602888d26199221d0d/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L225

5. That `elasticsearch.config` object is initialized here, from a `report.Settings` object which contains the UUID set in the Beat's configuration: https://github.com/ycombinator/beats/blob/1df1cf41d1de7fde07472e602888d26199221d0d/libbeat/monitoring/report/elasticsearch/elasticsearch.go#L121

6. So the check and assignment seen on this line is there to take the UUID from the `report.Settings` object and set it on the `elasticsearch.config` object.

> Better set this in the initializer directly then.

Could you point me to where this would be?

Reply:
Sorry, for causing confusion. It's mostly a small nit, but somewhat confusing at this place.

When creating `c`, `c.ClusterUUID == ""` by default. The `if` clause guarantees that if `setttings.ClusterUUID == ""`, then we keep `c.ClusterUUID == ""`. If not we set it to `setttings.ClusterUUID`. This makes the conditional a noop, because the post-condition of the code block is always `c.ClusterUUID == settings.ClusterUUID`. 

I meant to create `c` like this (at line 80):

```
  c := config{
     ... 
      ClusterUUID: settings.ClusterUUID,
     ...
  }
```

Reply:
Ugh, you're so right. I totally missed that 🤦‍♂. Thanks for catching it. Fixed in 740059e88.